lib/Controller/DocumentRevisionController.php

```diff
@@ -0,0 +1,167 @@
+<?php
+/**
+ * @author Piotr Mrowczynski <piotr@owncloud.com>
+ *
+ * @copyright Copyright (c) 2023, ownCloud GmbH
+ * @license AGPL-3.0
+ *
+ * This code is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License, version 3,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License, version 3,
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>
+ *
+ */
+namespace OCA\Richdocuments\Controller;
+
+use OCP\AppFramework\Controller;
+use OCP\AppFramework\Http;
+use OCP\AppFramework\Http\JSONResponse;
+use OCP\Files\InvalidPathException;
+use OCP\Files\IRootFolder;
+use OCP\Files\Storage\IVersionedStorage;
+use OCP\IRequest;
+use OCP\IUserSession;
+
+class DocumentRevisionController extends Controller {
+	/**
+	 * @var IUserSession The user session service
+	 */
+	private $userSession;
+
+	/**
+	 * @var IRootFolder The root folder service
+	 */
+	private $rootFolder;
+
+	public function __construct(
+		string $appName,
+		IRequest $request,
+		IUserSession $userSession,
+		IRootFolder $rootFolder
+	) {
+		parent::__construct($appName, $request);
+		$this->userSession = $userSession;
+		$this->rootFolder = $rootFolder;
+	}
+
+	/**
+	 * Get collabora document revisions for:
+	 * - the base template if fileId is null
+	 * - file in user folder (also shared by user/group) if fileId not null
+	 *
+	 * @NoAdminRequired
+	 */
+	public function list($fileId) {
+		if (\is_numeric($fileId)) {
+			// parse fileId pointing to file
+			$fileId = (int) $fileId;
```

Comment:
we might need to check the behavior with unreasonable big numbers. It might be ok if we document that numbers bigger than 2⁶⁴ will be considered as the maximum integer value (if that's the behavior)
As long as the behavior is ok, the documentation should be enough.

Reply:
we have this handling everywhere in WOPI. lets keep it like is, I know suboptimal